### ISSUE TITLE
MSIL compiler - add support for C++

### ISSUE THREAD

**Issue by GitMensch** [LADSoft/OrangeC#371]
As brought up to #369 this should be make possible some day, to be able to compile it with itself, creating a MSIL compiler that runs as MSIL

This is *not* intended to be any priority, just "a ticket to remember".

**Comment by chuggafan** [comment]
http://www.ecma-international.org/publications/files/ECMA-ST/ECMA-372.pdf In case you're wondering btw, this is the PDF of the C++/CLI spec. It'd be awesome to try to get a second C++/CLI compiler in existence, but it's still a big project to undertake.

**Comment by LADSoft** [comment]
yeah this is probably the lowest priority thing I've got to do :).

FWIW I had never used C++/CLI before I wrote the msil compiler, so it wouldn't be conformant with even the C portions.   Mostly it is a C compiler with enough C++ (for example namespaces and simple classes) to be able to compile straight C programs while importing managed code...

**Comment by LADSoft** [comment]
I found out today that while C++/CLI syntax is standardized, Microsoft isn't supporting it in relation to the latest push to go to using .NET core 2.0.    Apparently the cost to support it is prohibitive...

**Comment by chuggafan** [comment]
Yhea, cost of support for more than 1 compiler would be fairly prohibitive, but as far as I can tell they're still supporting it on windows. It's more of a side thing in all honesty than something we should absolutely focus on. But getting to that standard syntax would probably be helpful for some people.

**Comment by bencz** [comment]
Hi David!
support for C ++/CLI will be removed soon :/

"C++/CLI is not mentioned in 'The .NET Language Strategy' by Mads 
Torgersen, posted February 1, 2017." ( 
https://en.wikipedia.org/wiki/C%2B%2B/CLI )
Look at this issue: https://github.com/dotnet/wpf/issues/607 ( search 
for C++ )

Alexandre S. Bencz

+55 {15} 997-980-511

**Comment by GitMensch** [comment]
@LADSoft wrote:
> I found out today that while C++/CLI syntax is standardized, Microsoft isn't supporting it in relation to the latest push to go to using .NET core 2.0. Apparently the cost to support it is prohibitive...

Yes, standardized as ECMA-372. From the quoted .NET Core 3.0 issue I see C++/CLI being better integrated and from the article on Wikipedia it seems available in the default .NET framework since VS 2005 (and newer, including 2019).

@bencz wrote:
> support for C ++/CLI will be removed soon :/

I have no clue how you come to this conclusion from the referenced links. Can you please elaborate?

**Comment by LADSoft** [comment]
yeah much later it turned out the conclusion was premature...   apparently there is going to be a small window where CLI isn't available from microsoft (although apparently GCC supports it lol) but microsoft is going to fully support it again within a couple of years.

**Comment by bencz** [comment]
@GitMensch .... is based on the links and also the compiler output ...
![image](https://user-images.githubusercontent.com/1174448/70317574-cacf4480-17fc-11ea-9cf1-90ca182b7f58.png)

**Comment by LADSoft** [comment]
hm you don't get those warnings if you compile with just /clr

**Comment by bencz** [comment]
I think it's a good idea to keep an eye on this issue: https://github.com/dotnet/coreclr/issues/659#issuecomment-570452066

Look at the last messages....

**Comment by GitMensch** [comment]
The most important seems to be MS announcement https://devblogs.microsoft.com/cppblog/the-future-of-cpp-cli-and-dotnet-core-3/

**Comment by LADSoft** [comment]
yeah seems like it might be nice to support this some day...   be quite a while based on the current workload before I could get to it though.    Sometimes I wish I had help...

**Comment by bencz** [comment]
Hi David!
What kind of help you need ?

**Comment by LADSoft** [comment]
@bencz, well ,write me a compiler?    lol no really there are a lot of feature requests and it will be forever until I get to them all, even help with some of the smaller issues would be nice...

I am going to write a detailed thing about how to port this as soon as I get past #297, maybe someone will want to pick up the powerpc port.    While that is the kind of thing I love to do myself it is probably also one of the easiest things for someone else to pick up lol!

**Comment by bencz** [comment]
https://devblogs.microsoft.com/cppblog/porting-a-c-cli-project-to-net-core/

**Comment by LADSoft** [comment]
thanks!
--------------
I was thinking again about writing documents about porting this thing...    it is a big deal though because of the assembler...
